core/unit_test/cuda/TestCuda_InterOp_StreamsMultiGPU.cpp

```diff
@@ -16,34 +16,158 @@
 
 #include <TestCuda_Category.hpp>
 #include <Test_InterOp_Streams.hpp>
+#include <memory>
 
 namespace {
-TEST(cuda, multi_gpu) {
-  Kokkos::initialize();
 
-  int n_devices;
-  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaGetDeviceCount(&n_devices));
+struct StreamsAndDevices {
+  std::shared_ptr<std::array<cudaStream_t, 2>> streams;
```

Comment:
Why did you go for a `shared_ptr` rather than other alternatives?
* `unique_ptr`
* Move cleanup logic to the class destructor and delete copy constructor and copy assignment operator
* GTest's native support for "fixtures"

I am not necessarily blocking but the approach looks weird to me

Reply:
- It was more convenient to use `shared_ptr` than `unique_ptr` with a custom deleter.
- could do class destructor + delete special member functions
- don't like messing with gtest fixtures

Reply:
> * It was more convenient to use `shared_ptr` than `unique_ptr` with a custom deleter.

How so?  I am not very familiar with the differences.

> * don't like messing with gtest fixtures

Why not?  My understanding is that it was precisely developed for that use case you have.

Reply:
> How so? I am not very familiar with the differences.

For `unique_ptr` the deleter type is part of the type but not for `shared_ptr`. Also, I don't have to care about using special member functions with a `shared_ptr` member.

> Why not? My understanding is that it was precisely developed for that use case you have.

I rather start from a more clean slate for the two unit tests.

In the end, these are all small differences. If you are unhappy with this approach, I can do whatever. I just don't think that would change/improve the quality of the unit test much.

Reply:
If you don't want to use GTest fixture, then please move the cleanup logic to the destructor and make sure to make it non-copyable.

Reply:
See 6440a25